astropy/utils/tests/test_data.py

```diff
@@ -152,17 +154,46 @@ def test_local_data_obj(filename):
             assert f.read().rstrip() == b'CONTENT'
 
 
-@pytest.mark.parametrize(('filename'), ['invalid.dat.gz', 'invalid.dat.bz2'])
-def test_local_data_obj_invalid(filename):
+@pytest.fixture(params=['invalid.dat.bz2', 'invalid.dat.gz'])
+def bad_compressed(request, tmpdir):
+
+    # These contents have valid headers for their respective file formats, but
+    # are otherwise malformed and invalid.
+    bz_content = b'BZhinvalid'
+    gz_content = b'\x1f\x8b\x08invalid'
+
+    filename = os.path.join(tmpdir, request.param)
```

Comment:
I think tmpdir is a py.path object or something, not a string, which is causing the fails.

Reply:
yes, the standard way to use `tmpdir` is something like that: `tmpdir.join(filename).strpath`

Reply:
Actually tmpdir even has its own write method you might be able to use. Hard to post more on phone right now.

Reply:
Ah, right, I always forget that this doesn't work for earlier versions of python and/or pytest (it works in my environment). Generally I have done `os.path.join(str(tmpdir), ...)`, which works in all environments.